app/models/manageiq/providers/openstack/inventory/parser/cloud_manager.rb

```diff
@@ -0,0 +1,458 @@
+    flavor[:swap_disk_size] = f.swap.to_i.megabytes
+    flavor[:ephemeral_disk_size] = f.ephemeral.nil? ? nil : f.ephemeral.to_i.gigabytes
+    flavor[:ephemeral_disk_count] = if f.ephemeral.nil?
+                                      nil
+                                    elsif f.ephemeral.to_i > 0
+                                      1
+                                    else
+                                      0
+                                    end
+    if f.is_public
+      # public flavors are associated with every tenant
+      flavor[:cloud_tenants] = collector.tenants.map { |tenant| persister.cloud_tenants.lazy_find(tenant.id) }
+    else
+      # Add tenants with access to the private flavor
+      flavor[:cloud_tenants] = collector.tenant_ids_with_flavor_access(f.id).map { |tenant_id| persister.cloud_tenants.lazy_find(tenant_id) }
+    end
+  end
+
+  def orchestration_stack_resources(stack)
+    resources = []
+    raw_resources = safe_list { stack.resources }
+    # reject resources that don't have a physical resource id, because that
+    # means they failed to be successfully created
+    raw_resources.reject! { |r| r.physical_resource_id.nil? }
+    raw_resources.each do |resource|
+      uid = resource.physical_resource_id
+      o = persister.orchestration_stacks_resources.find_or_build(uid)
+      o[:ems_ref] = uid
+      o[:logical_resource] = resource.logical_resource_id
+      o[:physical_resource] = resource.physical_resource_id
+      o[:resource_category] = resource.resource_type
+      o[:resource_status] = resource.resource_status
+      o[:resource_status_reason] = resource.resource_status_reason
+      o[:last_updated] = resource.updated_time
+      o[:stack_id] = stack.id
+      s = persister.vms.find_or_build(uid)
+      s[:orchestration_stack] = persister.orchestration_stacks.lazy_find(stack.id)
+
+      resources << o
+    end
+    resources
+  end
+
+  def orchestration_stack_parameters(stack)
+    parameters = []
+    raw_parameters = safe_list { stack.parameters }
+    raw_parameters.each do |param_key, param_val|
+      uid = compose_ems_ref(stack.id, param_key)
+      o = persister.orchestration_stacks_parameters.find_or_build(uid)
+      o[:ems_ref] = uid
+      o[:name] = param_key
+      o[:value] = param_val
+      o[:stack_id] = stack.id
+      parameters << o
+    end
+    parameters
+  end
+
+  def orchestration_stack_outputs(stack)
+    outputs = []
+    raw_outputs = safe_list { stack.outputs }
+    raw_outputs.each do |output|
+      uid = compose_ems_ref(stack.id, output['output_key'])
+      o = persister.orchestration_stacks_outputs.find_or_build(uid)
+      o[:ems_ref] = uid
+      o[:key] = output['output_key']
+      o[:value] = output['output_value']
+      o[:description] = output['description']
+      o[:stack_id] = stack.id
+      outputs << o
+    end
+    outputs
+  end
+
+  def orchestration_template(stack)
+    template = safe_call { stack.template }
```

Comment:
also extracting to the Collector, best would be if Parser does not do any API call by itself